graspologic/plot/plot.py

```diff
@@ -671,6 +673,258 @@ def pairplot(
     return pairs
 
 
+def _plot_ellipse_and_data(
+    data, X, j, k, means, covariances, ax, label_palette, cluster_palette, alpha
+):
+    r"""
+    plot_ellipse makes a scatter plot from the two dimensions j,k where j
+    corresponds to x-axis
+    and k corresponds to the y-axis onto the axis that is ax. plot_ellipse then
+    applies a gmm ellipse onto the scatterplot
+    using the data from Y_(which is stored in data["clusters"]),
+    means, covariances.
+
+    Parameters
+    ----------
+    X : array-like, shape (n_samples, n_features)
+        Input data.
+    j : int
+        column index of feature of interest from X which will be the x-axis data.
+    k : int
+        column index of feature of interest from X which will be the y-axis data.
+    means : array-like, shape (n_components, n_features)
+        Estimated means from gmm
+    covariances : array-like, shape (with 'full') (n_components, n_features, n_features)
```

Comment:
@bdpedigo I see from the code below and this line that plotting ellipses for a GM model with a cov_type other that "full" is not supported. But I think it could be helpful if it's applicable for all cov_types. And I do see that there is the code to reformat `.covariances_` in your old [version](https://github.com/neurodata/maggot_models/blob/a6ffe01110dd319b7763d4ba2e8c5fe3039357e7/notebooks/179.0-BDP-mb-clustering.py#L40) and in the reference listed below. So if I'm not mistaken what is the reason for excluding that?

Reply:
you are correct, I must have missed this when reviewing, should be fixed.

Reply:
I actually have added a piece of code myself for plotting but then realized you already have that in yours. Anyway, I can make a PR soon.

Reply:
great!